File: ios_storyboard/TodoApp/Features/Todo/Todo.swift
import Foundation

struct Todo: Identifiable, Hashable, Sendable {
    let id: String
    var title: String
    var description: String
    var isCompleted: Bool
    var priority: TodoPriority
    var createdAt: Date

    init(
        id: String = UUID().uuidString,
        title: String,
        description: String = "",
        isCompleted: Bool = false,
        priority: TodoPriority = .medium,
        createdAt: Date = .now
    ) {
        self.id = id; self.title = title; self.description = description
        self.isCompleted = isCompleted; self.priority = priority; self.createdAt = createdAt
    }
}

enum TodoPriority: String, CaseIterable, Sendable {
    case low = "low", medium = "medium", high = "high"

    var label: String {
        switch self { case .low: "低"; case .medium: "中"; case .high: "高" }
    }
}

enum TodoFilter: String, CaseIterable {
    case all = "すべて", active = "未完了", completed = "完了"
}

extension Todo {
    static let samples: [Todo] = [
        Todo(id: "1", title: "牛乳を買う", description: "近くのスーパーで", priority: .medium),
        Todo(id: "2", title: "コードレビュー", description: "PRを確認する", isCompleted: true, priority: .high),
        Todo(id: "3", title: "ランニング", description: "30分", priority: .low),
    ]
}

File: android_compose/app/src/main/java/com/example/myapplication/features/todo/Todo.kt
package com.example.myapplication.features.todo

data class Todo(
    val id: String,
    val title: String,
    val description: String = "",
    val isCompleted: Boolean = false,
    val priority: TodoPriority = TodoPriority.MEDIUM,
    val createdAt: Long = System.currentTimeMillis(),
)

enum class TodoPriority { LOW, MEDIUM, HIGH }
enum class TodoFilter { ALL, ACTIVE, COMPLETED }

val dummyTodos = listOf(
    Todo(id = "1", title = "Jetpack Compose を学ぶ", description = "公式ドキュメントから始める", priority = TodoPriority.HIGH),
    Todo(id = "2", title = "Room で永続化する", description = "Entity / Dao / Database の3層", isCompleted = true, priority = TodoPriority.MEDIUM),
    Todo(id = "3", title = "Navigation Compose で画面遷移", priority = TodoPriority.LOW),
    Todo(id = "4", title = "Ktor Client で API 通信", priority = TodoPriority.MEDIUM),
    Todo(id = "5", title = "OAuth フローを実装する", priority = TodoPriority.HIGH),
)
